utils/phased_barrier.hh

```diff
@@ -71,6 +74,16 @@ public:
         return old_gate->close().then([old_gate, op = start()] {});
     }
 
+    // Ends the current phase but doesn't start a new one. Any new attempts at
+    // starting an operation will result in `gate_closed` exception.
+    future<> close() noexcept {
+        return _gate->close();
+    }
+
+    bool is_closed() const noexcept {
+        return _gate->is_closed();
+    }
+
```

Comment:
Please add a dedicated regression test for this. phased_barrier is tricky and a regression test would make further changes safer.